### PR TITLE
Extended component highlighter

### PR DESCRIPTION
**Pull request by KutnerUri**
## Proposed Changes

- new design (rebuild label)
- show additional components associated with the DOM element
- componentRule filter (should have already been merged)

### CODE REVIEW

File: scopes/react/component-highlighter/label/component-strip.module.scss [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+  &[data-size='m'] {
+    font-size: 14px;
+    line-height: 16px;
+  }
+
+  &[data-size='l'] {
+    font-size: 16px;
+    line-height: 24px;
+  }
```
Comment (odedre) [inline scopes/react/component-highlighter/label/component-strip.module.scss:25]: so we went with the hardcoded sizes?

File: scopes/react/component-highlighter/label/label.tsx [hunk 2]
```diff
@@ -1,70 +1,39 @@
...
-  flip?: boolean;
-  /** continually update label position to match moving elements */
-  watchMotion?: boolean;
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { ComponentMetaHolder } from '@teambit/react.ui.highlighter.component-metadata.bit-component-meta';
+
+import styles from './label.module.scss';
+import { ComponentStrip, ComponentStripSize } from './component-strip';
+import { OtherComponentsPopper } from './other-components';
```
Comment (odedre) [inline scopes/react/component-highlighter/label/label.tsx:7]: consider renaming. `OtherComponentsPopper` is confizing

File: scopes/react/component-highlighter/label/other-components.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+export type OtherComponentsProps = {
+  components: ComponentMetaHolder[];
+  size?: ComponentStripSize;
+  start?: number;
+  end?: number;
+} & TippyProps;
+
+// a popper ("tooltip") that shows the additional React Components related to this dom element
+export function OtherComponentsPopper({
```
Comment (odedre) [inline scopes/react/component-highlighter/label/other-components.tsx:16]: maybe `relatedComponents`?

**Review by odedre — APPROVED** [review]

File: scopes/react/component-highlighter/label/other-components.tsx (hunk 3 above)
Comment (KutnerUri) [reply scopes/react/component-highlighter/label/other-components.tsx:16]: I like!

File: scopes/react/component-highlighter/label/component-strip.module.scss (hunk 1 above)
Comment (KutnerUri) [reply scopes/react/component-highlighter/label/component-strip.module.scss:25]: yes

File: scopes/react/component-highlighter/label/label.tsx (hunk 2 above)
Comment (KutnerUri) [reply scopes/react/component-highlighter/label/label.tsx:7]: ok

**Pull request merged**